### ISSUE TITLE
False positive for shell task with with_items: [305] Use shell only when shell functionality is required

### ISSUE THREAD

**Issue by thorge** [ansible-community/ansible-lint#662]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.9.4
ansible-lint 4.2.0
```

- ansible installation method: OS package
- ansible-lint installation method: pip

# Desired Behaviour

If you have a task where the shell is used in a loop, the linter should recognize the stream operations like <, >, | and &. If env vars are used, it should also detect. The [UseCommandInsteadOfShellRule.py](https://github.com/ansible/ansible-lint/blob/master/lib/ansiblelint/rules/UseCommandInsteadOfShellRule.py) searches for `&|<>;$\n*[]{}?` but not in loop items.

```
- name: Task
  shell: "{{ item }}"
  with_items:
    - echo 'foo' > bar 
  changed_when: false
```

# Actual Behaviour (Bug report only)

Linting returns: `[305] Use shell only when shell functionality is required`, although shell functionality is required in loop items.

**Comment by ssbarnea** [comment]
Marking this as wontfix because we cannot go so far with linting, as this can only be identified at runtime.
